src/ucm/util/sys.c

```diff
@@ -30,6 +32,8 @@
 
 #define UCM_PROC_SELF_MAPS "/proc/self/maps"
 
+ucs_status_t (*ucm_mem_type_get_current_device_info[UCS_MEMORY_TYPE_LAST])(ucs_sys_bus_id_t *bus_id);
```

Comment:
let's add a "get_device_bus_id" method to ucm_event_installer_t

Reply:
cc @bureddy 

Does adding `get_device_bus_id` make sense? As I see it, current `ucm_event_installer_t` methods like `.get_existing_alloc` and `.install` help capture existing and future memtype alloc/free events in order to update `memtype_cache`. The similarity here is that `get_device_bus_id` could be used to capture context create/destroy events using new UCM_EVENT flags. This would mean we would are looking at intercepting cuCtxCreate/Destroy and cudaSetDevice/cudaDeviceReset and updating current context for cuda memtype.  @yosefe Is this the change you're suggesting?

Later the current context and associated bus_id would be queried using [`get_mem_type_current_device_info`](https://github.com/openucx/ucx/pull/5299/files#diff-fa26066bd292c6995d832018e17a0481R101) in IB UCT as a temporary workaround for distance querying. If context create/destroy events are going to used going forward for other reasons (which I think makes sense) then it's better to create a separate PR for that and use it here. But if it's only for the temporary workaround, I'm not sure if the above effort is justified. WDYT?

Reply:
we already have a list of func ptrs for memtype components: ucm_event_installer_t
so no need to define a new array ucm_mem_type_get_current_device_info. instead add new func ptr to ucm_event_installer_t struct.

Reply:
I guess we would add something along the following lines:

```diff
diff --git a/src/ucm/cuda/cudamem.c b/src/ucm/cuda/cudamem.c
index d95855cb8..00198c1ab 100644
--- a/src/ucm/cuda/cudamem.c
+++ b/src/ucm/cuda/cudamem.c
@@ -508,7 +508,8 @@ ucs_status_t ucm_cuda_get_current_device_info(ucs_sys_bus_id_t *bus_id)
 static ucm_event_installer_t ucm_cuda_initializer = {
     .install            = ucm_cudamem_install,
-    .get_existing_alloc = ucm_cudamem_get_existing_alloc
+    .get_existing_alloc = ucm_cudamem_get_existing_alloc,
+    .get_device_info    = ucm_cuda_get_current_device_info
 };
```

If we add the above function pointer to ucm_event_installer_t we have to go through event_installer_list, detect which one should be used for the memory type of interest and then call that function through event interface. Something along the following lines in the implementation of [ucm_get_mem_type_current_device_info](https://github.com/openucx/ucx/pull/5299/files#diff-fa26066bd292c6995d832018e17a0481R101):
```
        ucs_list_for_each(event_installer, &ucm_event_installer_list, list) {
            if (mem_type == ucm_get_event_installer_type(event_installer)) {
                return event_installer->get_device_info(&bus_id);
            }
        }
```

Is that the suggestion?

Reply:
maybe get_device_info can accept mem_type. If not the right type, it will return ERR_UNSUPPORTED

Reply:
Can make that change.

Reply:
I've updated the PR with the above suggestions.